File: test/test_matlab_python_interop.py
import math
import numpy as np
import sys
import time

from dime import DimeClient

d = DimeClient("ipc", sys.argv[1]);
d.join("python");

while "matlab" not in d.devices():
    time.sleep(0.05)

time.sleep(1)

d.sync()

assert d["nothing"] is None
assert d["boolean"] is True
assert d["float"] == math.pi
assert d["int"] == 0xDEADBEEF
assert d["complexfloat"] == complex(math.sqrt(0.75), 0.5)
assert np.array_equal(d["matrix"], np.array([[1, 2, 3], [4, 5, 6], [7, 8, 9.01]]))
assert d["string"] == "Hello world!"
assert d["sequence"] == [-1, ":)", False, None]
assert d["mapping"] == {"foo": 2, "bar": "Green eggs and spam"}

d.send("matlab", "nothing", "boolean", "int", "float", "complexfloat", "matrix", "string", "sequence", "mapping");

File: test/test_matlab_python_interop.m
function [] = test_matlab_python_interop(pathname)
    d = dime('ipc', pathname);

    d.join('matlab');

    while ~any(strcmp(d.devices(), 'python'))
        pause(0.05);
    end

    nothing = [];
    boolean = true;
    int = int64(0xDEADBEEF);
    float = pi;
    complexfloat = complex(sqrt(0.75), 0.5);
    matrix = [1, 2, 3; 4, 5, 6; 7, 8, 9.01];
    string = 'Hello world!';
    sequence = {int64(-1), ':)', false, []};
    mapping = struct('foo', int64(2), 'bar', 'Green eggs and spam');

    d.send('python', 'nothing', 'boolean', 'int', 'float', 'complexfloat', 'matrix', 'string', 'sequence', 'mapping');

    nothing_old = nothing;
    clear nothing;
    boolean_old = boolean;
    clear boolean;
    int_old = int;
    clear int;
    float_old = float;
    clear float;
    complexfloat_old = complexfloat;
    clear complexfloat;
    matrix_old = matrix;
    clear matrix;
    string_old = string;
    clear string;
    sequence_old = sequence;
    clear sequence;
    mapping_old = mapping;
    clear mapping;

    pause(2);

    d.sync();

    assert(isequal(nothing_old, nothing));
    assert(isequal(boolean_old, boolean));
    assert(isequal(float_old, float));
    assert(isequal(int_old, int));
    assert(isequal(complexfloat_old, complexfloat));
    assert(isequal(matrix_old, matrix));
    assert(isequal(string_old, string));
    assert(isequal(sequence_old, sequence));
    assert(isequal(mapping_old, mapping));
end
